### DISCUSSION TITLE
NotImplementedError with column.type.python_type

### DISCUSSION THREAD

**Discussion by freis** [tiangolo/pydantic-sqlalchemy#99] [Questions] [answered]
While testing your project I ran into this issue. Not sure where the column.type.python_test exists because when I was checking the sqlalchemy table declaration there was no reference to it.

```
Traceback (most recent call last):
  File "app/db/tables.py", line 31, in <module>
    PydanticTest = sqlalchemy_to_pydantic(Test)
  File "/home/vagrant/venv/project/lib/python3.8/site-packages/pydantic_sqlalchemy/main.py", line 21, in sqlalchemy_to_pydantic
    python_type = column.type.python_type
  File "/home/vagrant/venv/project/lib/python3.8/site-packages/sqlalchemy/sql/type_api.py", line 409, in python_type
    raise NotImplementedError()
NotImplementedError
```

**Comment by freis** [comment]
Reading more into the issue it seems that the problem is with my UUID column, and it wasn't able to translate the UUID because it had no python_type in the UUID type it seems.

```
from sqlalchemy import Column, func
from sqlalchemy.dialects.postgresql import UUID
test_id = Column(UUID, primary_key=True, server_default=func.uuid_generate_v1mc())
```

Would changing the function sqlalchemy_to_pydantic to do the follow someting acceptable or even correct?
```
import uuid
python_type = uuid.UUID if str(column.type) == 'UUID' else column.type.python_type
```

**Comment by AntonyM71** [comment]
Give how common that UUID's are inn databases these days, it's a shame that this isn't handled properly.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :sweat_smile:

This would be a feature request.

I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things. :rocket:
